source/common/common/regex.cc

```diff
@@ -47,12 +49,54 @@ class CompiledGoogleReMatcher : public CompiledMatcher {
       throw EnvoyException(regex_.error());
     }
 
-    const uint32_t max_program_size =
-        PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.google_re2(), max_program_size, 100);
-    if (static_cast<uint32_t>(regex_.ProgramSize()) > max_program_size) {
-      throw EnvoyException(fmt::format("regex '{}' RE2 program size of {} > max program size of "
-                                       "{}. Increase configured max program size if necessary.",
-                                       config.regex(), regex_.ProgramSize(), max_program_size));
+    const uint32_t regex_program_size = static_cast<uint32_t>(regex_.ProgramSize());
+
+    // Check if the deprecated field max_program_size is set first, and follow the old logic if so.
+    if (config.google_re2().has_max_program_size()) {
+      const uint32_t max_program_size =
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.google_re2(), max_program_size, 100);
+      if (static_cast<uint32_t>(regex_program_size > max_program_size)) {
```

Comment:
This static_cast looks wrong. Why would you static cast a bool expression to a uint32_t as an if-condition?

Did you mean: `if (static_cast<uint32_t>(regex_program_size) > max_program_size) ...` ?

Reply:
you're right, the parentheses were misplaced when I changed the var name